src/sagemaker/jumpstart/curated_hub/sync/request.py

```diff
@@ -0,0 +1,148 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module provides a class that perfrms functionalities similar to ``S3:Copy``."""
+from __future__ import absolute_import
+from dataclasses import dataclass
+from typing import Generator, List
+
+from botocore.compat import six
+
+from sagemaker.jumpstart.curated_hub.sync.comparator import BaseComparator
+from sagemaker.jumpstart.curated_hub.types import FileInfo, S3ObjectLocation
+
+advance_iterator = six.advance_iterator
+
+
+@dataclass
+class HubSyncRequest:
+    """HubSyncRequest class"""
+
+    files: List[FileInfo]
+    destination: S3ObjectLocation
+
+    def __init__(
+        self, files_to_copy: Generator[FileInfo, FileInfo, FileInfo], destination: S3ObjectLocation
+    ):
+        """Contains information required to sync data into a Hub.
+
+        Returns:
+            :var: files (List[FileInfo]): Files that shoudl be synced.
+            :var: destination (S3ObjectLocation): Location to which to sync the files.
```

Comment:
nit: this is unusual, strictly speaking `init` doesn't return anything. Please try to stay consistent with the rest of the docbase w.r.t constructor docstring, as in no `:var:`

Also, typo on line 39 (shoudl)